snapd-glib/snapd-task-data.c

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (C) 2024 Canonical Ltd.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2 or version 3 of the License.
+ * See http://www.gnu.org/copyleft/lgpl.html the full text of the license.
+ */
+
+#include <string.h>
+
+#include "snapd-task-data.h"
+
+/**
+ * SECTION: snapd-task-data
+ * @short_description: Task extra data
+ * @include: snapd-glib/snapd-glib.h
+ *
+ * A #SnapdTaskData contains extra information on a task in a #SnapdChange.
+ */
+
+/**
+ * SnapdTaskData:
+ *
+ * #SnapdTaskData contains extra information for a task in a Snap transaction.
+ *
+ * Since: 1.66
+ */
+
+struct _SnapdTaskData
+{
+    GObject parent_instance;
+    GPtrArray *affected_snaps;
+};
+
+enum
+{
+    PROP_ID = 1,
+    PROP_AFFECTED_SNAPS,
+    PROP_LAST
+};
+
+G_DEFINE_TYPE (SnapdTaskData, snapd_task_data, G_TYPE_OBJECT)
+
+
+/**
+ * snapd_task_data_get_affected_snaps:
+ * @task_data: a #SnapdTaskData.
+ *
+ * Get the list of snaps that are affected by this task, or %NULL if snapd doesn't send this data.
+ *
+ * Returns: (transfer full) (allow-none) (element-type gchar*): a #GPtrArray or %NULL.
+ *
+ * Since: 1.66
+ */
+GPtrArray *
```

Comment:
In other cases (e.g. SnapdConnection.get_slot_attribute_names) we've returned GStrv here. I think for consistency we should do the same here.

Reply:
You are right. Done.